### PR TITLE
[TD-13119]<docs>:csharp connector add readme

### PR DESCRIPTION
**Pull request by xleili**

### CODE REVIEW

File: src/connector/C#/README.md [hunk 1]
```diff
@@ -0,0 +1,440 @@
+## CSharp Connector
+
+* The C # connector supports: Linux 64/Windows x64/Windows x86.
+* C# connector can be download and include as normal table form [Nuget.org](https://www.nuget.org/packages/TDengine.Connector/).
+
+### Installation preparation
+
+* Install TDengine client, please refer to the[steps of installing connector driver](https://www.taosdata.com/en/documentation/connector#driver).
+* .NET interface file TDengineDrivercs.cs and reference sample TDengineTest.cs
```
Comment (zitsen) [inline src/connector/C#/README.md]: `TDengineDrivercs` you mean `TDengineDriver`?

File: src/connector/C#/README.md [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+ are both located in the Windows client path:install_directory/examples/C#.
+* Install [.NET SDK](https://dotnet.microsoft.com/download)
+
+### Installation verification
+
+Run {client_installation_directory}/examples/C#/C#Checker/C#Checker.cs
+Need install .Net SDK first
```
Comment (zitsen) [inline src/connector/C#/README.md]: if this is new paragraph, finish previous paragraph with `.`, and add a break line.

File: src/connector/C#/README.md [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+
+```cmd
+cd {client_install_directory}/examples/C#/C#Checker
```
Comment (zitsen) [inline src/connector/C#/README.md]: `#` may cause error, use `\#` or "<path>" style.

File: src/connector/C#/README.md [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+// ... do something ...
+  string host = "127.0.0.1" ;
+  string configDir =  "C:/TDengine/cfg"; // For linux should be /etc/taos.
+  string user = "root";
+  string password = "taosdata";
+  string db = ''; // Also can set to the db name you want to connect.
+  string port = 0
+
+/* Set client options (optional step):charset,locale,timezone.
+ * Defualt: charset,locale,timezone same to system.
```
Comment (zitsen) [inline src/connector/C#/README.md]: Default.

File: src/connector/C#/README.md [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+ * Current support options:TSDB_OPTION_LOCALE,TSDB_OPTION_CHARSET, TSDB_OPTION_TIMEZONE,TSDB_OPTION_CONFIGDIR
+*/
+TDengine.Options((int)TDengineInitOption.TDDB_OPTION_CONFIGDIR,configDir);
```
Comment (zitsen) [inline src/connector/C#/README.md]: TDDB ?-> TSDB

File: src/connector/C#/README.md [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+   Console.WriteLine("Connect to TDengine success");
+}
+
+// Close TDengine Connection
+if (conn != IntPtr.Zero)
+{
+    TDengine.Close(this.conn);
+}
+
+// Recommand to clean environment, before exit the your application.
```
Comment (zitsen) [inline src/connector/C#/README.md]: Recommand -> Recommend

File: src/connector/C#/README.md [hunk 7]
```diff
@@ -0,0 +1,440 @@
+## CSharp Connector
+
+* The C # connector supports: Linux 64/Windows x64/Windows x86.
```
Comment (sangshuduo) [inline src/connector/C#/README.md]: C#, not 'C #'

File: src/connector/C#/README.md [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+    // be got from previous "Connection" sample
+    IntPrt res = TDengine.Query(conn, sql);
+    if ((res == IntPtr.Zero) || (TDengine.ErrorNo(res) != 0))
+    {
+         Console.Write(sql.ToString() + " failure, ");
+         if (res != IntPtr.Zero)
+         {
+             Console.Write("reason: " + TDengine.Error(res));
+         }
+         // Execute query sql faild
```
Comment (zitsen) [inline src/connector/C#/README.md]: faild -> failed

File: src/connector/C#/README.md [hunk 9]
```diff
@@ -0,0 +1,440 @@
+## CSharp Connector
+
+* The C # connector supports: Linux 64/Windows x64/Windows x86.
+* C# connector can be download and include as normal table form [Nuget.org](https://www.nuget.org/packages/TDengine.Connector/).
```
Comment (sangshuduo) [inline src/connector/C#/README.md]: downloaded, included

File: src/connector/C#/README.md [hunk 10]
```diff
@@ -0,0 +1,440 @@
...
+         {
+             Console.Write("reason: " + TDengine.Error(res));
+         }
+         // Execute query sql faild
+         // ... do something ...
+    }
+
+    // Retrieve data success,iterate through the "res"
+
+    // Feild count, num of field, that is equal to retrieved column number
```
Comment (zitsen) [inline src/connector/C#/README.md]: Feild -> Field

File: src/connector/C#/README.md [hunk 11]
```diff
@@ -0,0 +1,440 @@
...
+    int fieldCount = TDengine.FieldCount(res);
+    Console.WriteLine("field count: " + fieldCount);
+
+    // Getting querying result's feild info as a List.
```
Comment (zitsen) [inline src/connector/C#/README.md]: feild -> field

File: src/connector/C#/README.md [hunk 12]
```diff
@@ -0,0 +1,440 @@
...
+    List<TDengineMeta> metas = TDengine.FetchFields(res);
+    for(int j = 0; j < metas.Count; j++)
+    {
+        TDengineMeta meta = (TDengineMeta)metas[j];
+        Console.WriteLine($"index:{j},type:{meta.type},typename:{meta.TypeName()},name:{meta.name},size:{meta.size}");
+    }
+
+    // Traverse data from retreive result
```
Comment (zitsen) [inline src/connector/C#/README.md]: retreive -> retrieve

File: src/connector/C#/README.md (hunk 1 above)
Comment (sangshuduo) [inline src/connector/C#/README.md]: TDengineDrivercs.cs ?

File: src/connector/C#/README.md [hunk 13]
```diff
@@ -0,0 +1,440 @@
...
+TAOS_MULTI_BIND[] mBinds = new TAOS_MULTI_BIND[5];
+
+mBinds[0] = TaosMultiBind.MultiBindTimestamp(tsArr);
+mBinds[1] = TaosMultiBind.MultiBindBool(boolArr);
+mBinds[4] = TaosMultiBind.MultiBindInt(intArr);
+mBinds[5] = TaosMultiBind.MultiBindBigint(longArr);
+mBinds[12] = TaosMultiBind.MultiBindBinary(binaryArr);
+
+// After using instance of TAOS_MULTI_BIND and TAOS_BIND,
+// need to free the allocated ummanaged memory.
```
Comment (zitsen) [inline src/connector/C#/README.md]: ummanaged -> unmanaged

File: src/connector/C#/README.md [hunk 14]
```diff
@@ -0,0 +1,440 @@
...
+TDengine.StmtBindParam(stmt, qparams);
+
+// Execute statement
+TDengine.StmtExecute(stmt);
+
+// Get the query result, for SELECT only.
+// User application should free it with API FreeResult at the end.
+IntPtr result = TDengine.StmtUseResult(stmt);
+
+// Tthis "result" cam be traversed as normal sql query result.
```
Comment (zitsen) [inline src/connector/C#/README.md]: Tthis -> This

File: src/connector/C#/README.md [hunk 15]
```diff
@@ -0,0 +1,440 @@
...
+// ... Do something with "result" ...
+
+// Free result,free the restsult.
```
Comment (zitsen) [inline src/connector/C#/README.md]: restsult -> result

File: src/connector/C#/README.md [hunk 16]
```diff
@@ -0,0 +1,440 @@
...
+TDengine.FreeResult(result);
+
+// Cause we use unmanaged memory, we need to free these memory after execute.
+TaosMultiBind.FreeBind(qparams);
+
+// Close stmt, release resource
+TDengine.StmtClose(stmt);
+```
+
+* Assert (samples about how to assert every step of stmt is successed or not)
```
Comment (zitsen) [inline src/connector/C#/README.md]: is successed -> has succeeded or is success

File: src/connector/C#/README.md [hunk 17]
```diff
@@ -0,0 +1,440 @@
...
+
+```C#
+// Specail StmtInit().
```
Comment (zitsen) [inline src/connector/C#/README.md]: Special?

File: src/connector/C#/README.md [hunk 18]
```diff
@@ -0,0 +1,440 @@
...
+ Console.WriteLine(sql.ToString() + " success");
+}
+```
+
+* More samples reference from [examples](https://github.com/taosdata/TDengine/tree/develop/tests/examples/C%23/).
+
+**Note:**
+
+* TDengine V2.0. 3.0 supports both 32-bit and 64-bit Windows systems,
+so when. NET project generates a .exe file, please select the correspond
```
Comment (zitsen) [inline src/connector/C#/README.md]: line breaks in list should be indented:

```markdown
* a long long long long 
  indented next line.
* text two
```

**Review by zitsen — CHANGES_REQUESTED** [review]
See my comments

File: src/connector/C#/README.md (hunk 7 above)
Comment (xleili) [reply src/connector/C#/README.md]: fixed

File: src/connector/C#/README.md (hunk 18 above)
Comment (xleili) [reply src/connector/C#/README.md]: great！

File: src/connector/C#/README.md [hunk 19]
```diff
@@ -0,0 +1,435 @@
+## CSharp Connector
+
+* This C# connector supports: Linux 64/Windows x64/Windows x86.
+* This C# connector can be downloaded and included as a normal package form [Nuget.org](https://www.nuget.org/packages/TDengine.Connector/).
```
Comment (sangshuduo) [inline src/connector/C#/README.md]: from...

File: src/connector/C#/README.md [hunk 20]
```diff
@@ -0,0 +1,435 @@
...
+
+```cmd
+cd {client_install_directory}/examples/C\#/C#Checker
+//run c#checker.cs
+dotnet run -- -h <FQDN>
+```
+
+### Example Source Code
+
+You can find examples under follow directions:
```
Comment (sangshuduo) [inline src/connector/C#/README.md]: directories?

File: src/connector/C#/README.md [hunk 21]
```diff
@@ -0,0 +1,435 @@
+## CSharp Connector
+
+* This C# connector supports: Linux 64/Windows x64/Windows x86.
+* This C# connector can be downloaded and included as a normal package form [Nuget.org](https://www.nuget.org/packages/TDengine.Connector/).
+
+### Installation preparations
+
+* Install TDengine client, please refers to the[steps of installing connector driver](https://www.taosdata.com/en/documentation/connector#driver).
```
Comment (zitsen) [inline src/connector/C#/README.md]: leave a space after the

File: src/connector/C#/README.md [hunk 22]
```diff
@@ -0,0 +1,435 @@
...
+* .NET interface file TDengineDriver.cs and reference samples both
```
Comment (zitsen) [inline src/connector/C#/README.md:9]: remove *reference*

File: src/connector/C#/README.md [hunk 23]
```diff
@@ -0,0 +1,435 @@
...
+
+* Install TDengine client, please refers to the[steps of installing connector driver](https://www.taosdata.com/en/documentation/connector#driver).
+* .NET interface file TDengineDriver.cs and reference samples both 
+  located under Windows client's installation path:install_directory/examples/C#.
+* Install [.NET SDK](https://dotnet.microsoft.com/download)
+
+### Installation verification
+
+Run {client_installation_directory}/examples/C#/C#Checker/C#Checker.cs
+Need to install .Net SDK first.
```
Comment (zitsen) [inline src/connector/C#/README.md]: .NET SDK is instructed in Installation preparations section, so this line is not necessary.

File: src/connector/C#/README.md [hunk 24]
```diff
@@ -0,0 +1,435 @@
...
+### Example Source Code
+
+You can find examples under follow directions:
+
+* {client_installation_directory}/examples/C#
+* [github C# example source code](https://github.com/taosdata/TDengine/tree/develop/tests/examples/C%23)
+
+**Tips:**
+"TDengineTest" is an example that includes some basic sample code like
+connect,query and so on.
```
Comment (zitsen) [inline src/connector/C#/README.md]: leave a space after `,` .

File: src/connector/C#/README.md [hunk 25]
```diff
@@ -0,0 +1,435 @@
...
+
+### Use C# connector
+
+#### **prepare**
+
+**tips:** Need to install .NET SDK first.
+
+* Create a dotnet project. (Using console project as an example.)
```
Comment (zitsen) [inline src/connector/C#/README.md]: Create a dotnet project (using ...).

File: src/connector/C#/README.md [hunk 26]
```diff
@@ -0,0 +1,435 @@
...
+
+* More samples reference from [examples](https://github.com/taosdata/TDengine/tree/develop/tests/examples/C%23/).
+
+**Note:**
+
+* TDengine V2.0. 3.0 supports both 32-bit and 64-bit Windows systems,
+  so when .NET project generates a .exe file, please select correspond
+  with "X86" or "x64" for the "Platform" under "Solution"/"Project".
+* This .NET interface has been verified in Visual Studio 2015/2017,
+  and other VS versions have not being verified yet.
```
Comment (zitsen) [inline src/connector/C#/README.md]: 1. no `and`
2. have not being -> have not been

File: src/connector/C#/README.md [hunk 27]
```diff
@@ -0,0 +1,435 @@
...
+{
+ Console.WriteLine(sql.ToString() + " success");
+}
+```
+
+* More samples reference from [examples](https://github.com/taosdata/TDengine/tree/develop/tests/examples/C%23/).
+
+**Note:**
+
+* TDengine V2.0. 3.0 supports both 32-bit and 64-bit Windows systems,
```
Comment (zitsen) [inline src/connector/C#/README.md]: remove the space in version string:

`V2.0. 3.0` -> `V2.0.3.0`

**Review by sangshuduo — COMMENTED** [review]
see my comments

**Review by sangshuduo — APPROVED** [review]
LGTM

**Review by zitsen — APPROVED** [review]
LGTM

**Pull request merged**